codestyle/druid-forbidden-apis.txt

```diff
@@ -1,2 +1,4 @@
 com.google.common.collect.MapMaker @ Create java.util.concurrent.ConcurrentHashMap directly
-com.google.common.collect.Maps#newConcurrentMap() @ Create java.util.concurrent.ConcurrentHashMap directly
\ No newline at end of file
+com.google.common.collect.Maps#newConcurrentMap() @ Create java.util.concurrent.ConcurrentHashMap directly
+com.google.common.util.concurrent.Futures.transform(com.google.common.util.concurrent.ListenableFuture<I>, com.google.common.util.concurrent.AsyncFunction<? super I,? extends O>) @ Use io.druid.java.util.common.concurrent.ListenableFutures#transformAsync
```

Comment:
It's still different from other signatures. I'm not sure which syntax is correct, so please check that those prohibitions work

Reply:
https://github.com/policeman-tools/forbidden-apis/wiki/SignaturesSyntax I'm trying this out more to see.

If the signature is this

```
com.google.common.util.concurrent.Futures#transform(com.google.common.util.concurrent.ListenableFuture, com.google.common.util.concurrent.AsyncFunction) @ Use io.druid.java.util.common.concurrent.ListenableFutures#transformAsync
```


And I change one of the lines to this

```java
    return Futures.transform(
        publishFuture,
        (com.google.common.util.concurrent.AsyncFunction<? super SegmentsAndMetadata, ? extends SegmentsAndMetadata>) sam -> {
          //(Function<? super SegmentsAndMetadata, ? extends SegmentsAndMetadata>) sam -> {
          synchronized (segments) {
            sequenceNames.forEach(segments::remove);
          }
          final SettableFuture<SegmentsAndMetadata> samfuture = SettableFuture.create();
          samfuture.set(sam);
          return samfuture;
        }
    );
```


then maven gives me this:

```
[ERROR] Forbidden method invocation: com.google.common.util.concurrent.Futures#transform(com.google.common.util.concurrent.ListenableFuture, com.google.common.util.concurrent.AsyncFunction) [Use io.druid.java.util.common.concurrent.ListenableFutures#transformAsync]
[ERROR]   in io.druid.segment.realtime.appenderator.StreamAppenderatorDriver (StreamAppenderatorDriver.java:278)
[ERROR] Scanned 1005 (and 1148 related) class file(s) for forbidden API invocations (in 0.63s), 1 error(s).

```

So I'll make a change here that is validated to work. I could have sworn I tried the one in this comment before, but apparently not. Now I have a validated one though.